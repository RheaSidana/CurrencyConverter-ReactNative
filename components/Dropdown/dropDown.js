import React, { useState } from "react";
import DropDown from "react-native-paper-dropdown";

function Dropdown(props) {
  const [showDropDown, setShowDropDown] = useState(false);
  const { label, currencyList, currency, setCurrency /*, assign*/ } = props;
  const clist = currencyList.map((curr, k) => {
    return {
      label: curr.name,
      value: k,
    };
  });

  return (
    <DropDown
      label={label}
      mode={"outlined"}
      visible={showDropDown}
      showDropDown={() => setShowDropDown(true)}
      onDismiss={() => setShowDropDown(false)}
      value={currency}
      setValue={setCurrency}
      list={clist}
    />
  );
}

export default Dropdown;
