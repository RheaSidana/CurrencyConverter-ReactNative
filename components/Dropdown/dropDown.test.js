import React from "react";
import renderer, { act } from "react-test-renderer";
import Dropdown from "./dropDown";

jest.mock("react-native-paper-dropdown", () => "MockDropDown");

describe("Dropdown", () => {
  test("renders correctly with props", () => {
    const currencyList = [{ name: "USD" }, { name: "EUR" }];
    const setCurrency = jest.fn();

    const component = renderer.create(
      <Dropdown
        label="Select Currency"
        currencyList={currencyList}
        currency={0}
        setCurrency={setCurrency}      
      />
    );

    expect(component.toJSON()).toMatchSnapshot();
  });

  test("opens dropdown on showDropDown callback", () => {
    const currencyList = [{ name: "USD" }, { name: "EUR" }];
    const setCurrency = jest.fn();

    const component = renderer.create(
      <Dropdown
        label="Select Currency"
        currencyList={currencyList}
        currency={0}
        setCurrency={setCurrency}
      />
    );

    const instance = component.root;
    act(()=>{
      instance.findByType("MockDropDown").props.showDropDown();
    })
    expect(instance.findByType("MockDropDown").props.visible).toBe(true);
  });

  test("dismisses dropdown on onDismiss callback", () => {
    const currencyList = [{ name: "USD" }, { name: "EUR" }];
    const setCurrency = jest.fn();
    const component = renderer.create(
      <Dropdown
        label="Select Currency"
        currencyList={currencyList}
        currency={0}
        setCurrency={setCurrency}
      />
    );

    const instance = component.root;

    act(()=>{
      instance.findByType("MockDropDown").props.showDropDown();
    })
    expect(instance.findByType("MockDropDown").props.visible).toBe(true);

    
    act(()=>{
      instance.findByType("MockDropDown").props.onDismiss();
    })
    expect(instance.findByType("MockDropDown").props.visible).toBe(false);
  });

  test("sets currency value on selection", () => {
    const currencyList = [{ name: "USD" }, { name: "EUR" }];
    const setCurrency = jest.fn();
    const component = renderer.create(
      <Dropdown
        label="Select Currency"
        currencyList={currencyList}
        currency={0}
        setCurrency={setCurrency}
      />
    );

    const instance = component.root;
    instance.findByType("MockDropDown").props.setValue(1);
    expect(setCurrency).toHaveBeenCalledWith(1);
  });
});
