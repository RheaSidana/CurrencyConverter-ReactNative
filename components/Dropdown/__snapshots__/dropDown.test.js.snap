// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown renders correctly with props 1`] = `
<MockDropDown
  label="Select Currency"
  list={
    [
      {
        "label": "USD",
        "value": 0,
      },
      {
        "label": "EUR",
        "value": 1,
      },
    ]
  }
  mode="outlined"
  onDismiss={[Function]}
  setValue={[MockFunction]}
  showDropDown={[Function]}
  value={0}
  visible={false}
/>
`;
